### PR TITLE
Bump trie-db and reference-trie crates to 0.16.0.

### PR DESCRIPTION
**Pull request by jimpo**
Bump from 0.15 to 0.16 after breaking changes to trie-db crate.

### CODE REVIEW

File: memory-db/Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,7 @@ heapsize = { version = "0.4", optional = true }
 parity-util-mem = { version = "0.2", default-features = false }
 hash-db = { path = "../hash-db", default-features = false, version = "0.15.2"}
 hashbrown = { version = "0.6.3", default-features = false, features = [ "ahash" ] }
+ahash = "0.2.18"
```
Comment (jimpo) [inline memory-db/Cargo.toml]: There's a compilation error with `0.2.17`, which is permitted by the `0.2.11` constraint in `hashbrown`.

File: memory-db/Cargo.toml [hunk 2]
```diff
@@ -11,6 +11,8 @@ heapsize = { version = "0.4", optional = true }
 parity-util-mem = { version = "0.2", default-features = false }
 hash-db = { path = "../hash-db", default-features = false, version = "0.15.2"}
 hashbrown = { version = "0.6.3", default-features = false, features = [ "ahash" ] }
+# There's a compilation error with ahash-0.2.17, which is permitted by the 0.2.11 constraint in hashbrown.
+ahash = "0.2.18"
```
Comment (ordian) [inline memory-db/Cargo.toml]: As it's not a direct dependency, shouldn't it be fixed in Cargo.lock though?

Comment (jimpo) [reply memory-db/Cargo.toml]: One typically only commits `Cargo.lock`s for applications, not for libraries.

Comment (ordian) [reply memory-db/Cargo.toml]: Right, I meant in end-user application's `Cargo.lock`, but it's not a big deal anyway.

**Pull request merged**